pkg/output/tui/progress.go

```diff
@@ -175,17 +175,23 @@ func (b bubbleProgressHandler) speedChange() (tea.Model, tea.Cmd) {
 }
 
 func (b bubbleProgressHandler) percentChange(event percentChange) (tea.Model, tea.Cmd) {
-	var cmds []tea.Cmd
-
+	cmds := make([]tea.Cmd, 0, 1)
 	cmds = append(cmds, b.prog.SetPercent(float64(event)))
 
 	if event >= 1.0 {
-		cmds = append(cmds, tea.Sequence(b.finalPause(), tea.Quit))
+		cmds = append(cmds, b.quitSignal())
 	}
 
 	return b, tea.Batch(cmds...)
 }
 
+func (b *BubbleProgress) quitSignal() tea.Cmd {
+	// The final pause is to give the progress bar a chance to finish its
+	// animation before quitting. Otherwise, it ends abruptly, and the user
+	// might not see the progress bar at 100%.
+	return tea.Sequence(b.finalPause(), tea.Quit)
```

Comment:
I still don't like the additional wait. If your UI has a lot of progress bars, this wait will be visible.

But I don't see a way, with the current API of bubbles' progress, to achieve it otherwise...

Reply:
I was experimenting with [ViewAs() func](https://pkg.go.dev/github.com/charmbracelet/bubbles/progress#Model.ViewAs). You could see the experiment here: https://github.com/cardil/knative-client-pkg/commit/5d19f314622770ea5f8bffdf4c509b3d14eddc3c

It worked. However, the output isn't what I was going for. The progress bar is always doubled. See:

```
$ go run ./cmd/ght download --destination /tmp/ghet golangci/golangci-lint
⛳️ Getting information about latest release Done
🎉 Found 2 matching assets for v1.57.1
📥 12 golangci-lint-1.57.1-linux-amd64.tar.gz  █████████████████████████░░  94% ⋮  20.09 MiB/s ⋮  10.55 MiB
📥 12 golangci-lint-1.57.1-linux-amd64.tar.gz  ███████████████████████████ 100% ⋮  20.09 MiB/s ⋮  10.55 MiB
📥 22 golangci-lint-1.57.1-checksums.txt       ░░░░░░░░░░░░░░░░░░░░░░░░░░░   0% ⋮   0.00 B/s   ⋮   5.13 KiB
📥 22 golangci-lint-1.57.1-checksums.txt       ███████████████████████████ 100% ⋮   0.00 B/s   ⋮   5.13 KiB
🔍🔍 Verifying checksum for golangci-lint-1.57.1-linux-amd64.tar.gz Done
✅ All checksums match the downloaded assets
📦 Extracting archive: golangci-lint-1.57.1-linux-amd64.tar.gz
🎯 golangci-lint  ██████████████████████████████████████████████████████░░  96% ⋮ 123.39 MiB/s ⋮  26.64 MiB
🎯 golangci-lint  ████████████████████████████████████████████████████████ 100% ⋮ 123.39 MiB/s ⋮  26.64 MiB
```
I think, there's no easy way to avoid the doubling this way. This is because, we print the line outside the control of the bubbletea renderer (which uses a special shell display mode).

Reply:
I think, there's no easy way to avoid the doubling this way. This is because, we print the line outside the control of the bubbletea renderer (which uses a special shell display mode).

Reply:
This is very similar to showing 100% of completeness. You just print `""` instead of the progress.

I don't think it is possible with the current API. When the `tea.Quit` command is handled, the rendered stops, and restores the terminal to standard output mode. So, you can't replace the line that was already written.

I think the bubbletea should allow defining the last `tea.Quit` state, that will be printed just before the exit, and without waiting for the animation loop to draw it.

Reply:
Tracked here: https://github.com/knative/client-pkg/issues/163